libsolidity/experimental/analysis/TypeInference.cpp

```diff
@@ -204,6 +214,8 @@ bool TypeInference::visit(TypeClassDefinition const& _typeClassDefinition)
 	solAssert(m_analysis.annotation<TypeClassRegistration>(_typeClassDefinition).typeClass.has_value());
 	TypeClass typeClass = m_analysis.annotation<TypeClassRegistration>(_typeClassDefinition).typeClass.value();
 	Type typeVar = m_typeSystem.typeClassVariable(typeClass);
+	unify(typeAnnotation(_typeClassDefinition.typeVariable()), typeVar, _typeClassDefinition.location());
```

Comment:
```suggestion
	unify(type(_typeClassDefinition.typeVariable()), typeVar, _typeClassDefinition.location());
```
I'm confused how this got back into the base branch. We did have that argument before and said we'd revert this to ``type``, just as it's done elsewhere in the code base and which makes a lot more sense (what the function gives you is the type of its argument - why would you want to focus attention on this being any kind of annotation? That's just weird and unnecessarily verbose)

Reply:
#14730